compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -838,8 +838,16 @@ private boolean checkFunctionTypeEquality(BInvokableType source, BInvokableType
         }
 
         for (int i = 0; i < source.paramTypes.size(); i++) {
-            if (target.paramTypes.get(i).tag != TypeTags.ANY
-                    && !equality.test(source.paramTypes.get(i), target.paramTypes.get(i), unresolvedTypes)) {
+            if (!equality.test(source.paramTypes.get(i), target.paramTypes.get(i), unresolvedTypes)) {
+                return false;
+            }
+        }
+
+        if (source.restType != null) {
```

Comment:
What if source.restType == null and target.restType != null

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/19976/commits/e597b121b663f7f4609915a506376e1619ca525e